external/storm-mongodb/src/main/java/org/apache/storm/mongodb/bolt/MongoInsertBolt.java

```diff
@@ -99,6 +107,29 @@ public MongoInsertBolt withOrdered(boolean ordered) {
         return this;
     }
 
+    public MongoInsertBolt withFlushIntervalSecs(int flushIntervalSecs) {
+        this.flushIntervalSecs = flushIntervalSecs;
+        return this;
+    }
+
+    @Override
+    public Map<String, Object> getComponentConfiguration() {
+        Map<String, Object> conf = super.getComponentConfiguration();
+        if (conf == null) {
+            conf = new Config();
+        }
+
+        if (conf.containsKey(Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS) && flushIntervalSecs == 0) {

```

Comment:
@HeartSaVioR I find that `storm-solr` has the same bug as you stated. And I filed a JIRA(https://issues.apache.org/jira/browse/STORM-1669) for that. Maybe this fix should be included in Storm 1.0 release.